### PR TITLE
REPLAY-1862 Increate batchSizeLimit to 10MB for SR feature

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

In this PR we are increasing the SR batch size limit to 10MB into the core to support large payloads containing embedded base64 images. This follows the same approach as in iOS introduced with the Image recording feature added lately. 

Please note that the SR intake unlike RUM accepts payloads up to 10MB size.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt [hunk 1]
```diff
@@ -211,6 +211,25 @@ internal class SessionReplayFeature constructor(
...
     internal companion object {
+
+        /**
+         * Session Replay storage configuration with the following parameters:
+         * * max item size = 4 MB,
+         * * max items per batch = 500,
+         * * max batch size = 10 MB, SR intake batch limit is 10MB
+         * * old batch threshold = 18 hours.
+         */
+        internal val STORAGE_CONFIGURATION: FeatureStorageConfiguration =
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt:222]: isn't it better to take `DEFAULT` and call `copy` here, only overriding what is needed for SR?

Comment (mariusc83) [reply features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt:222]: Hmmm....good question, not sure which one is better. I think I will do that.

**Review by 0xnm — APPROVED** [review]

**Review by 0xnm — APPROVED** [review]

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt [hunk 2]
```diff
@@ -211,6 +211,17 @@ internal class SessionReplayFeature constructor(
...
+            FeatureStorageConfiguration.DEFAULT.copy(maxBatchSize = 10 * 1024 * 1024)
```
Comment (xgouchet) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt:223]: Shouldn't the maxItemSize be increased as well?

**Review by xgouchet — APPROVED** [review]

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt (hunk 2 above)
Comment (mariusc83) [reply features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt:223]: I am actually assessing this...I think 512KB should be enough...let me check with Maciej

Comment (mariusc83) [reply features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt]: Yes, you were right, they increased that too so right now we are 10MB, 10MB for both item and bath size to align with them for SR

**Review by jonathanmos — APPROVED** [review]

**Pull request merged**